core-web/libs/portlets/edit-ema/portlet/src/lib/edit-ema-editor/components/ema-contentlet-tools/ema-contentlet-tools.component.ts

```diff
@@ -213,6 +211,25 @@ export class EmaContentletToolsComponent implements OnInit, OnChanges {
         };
     }
 
+    /**
+     * Check if Contentlet or Container are changed
+     *
+     * @private
+     * @param {SimpleChange} contentletChange
+     * @return {*}  {boolean}
+     * @memberof EmaContentletToolsComponent
+     */
+    private hasContentletOrContainerChanged(contentletChange: SimpleChange): boolean {
+        const currentValue = contentletChange.currentValue?.payload;
+        const previousValue = contentletChange.previousValue?.payload;
+
+        const hasIdentifierChanged =
+            currentValue?.contentlet.identifier !== previousValue?.contentlet.identifier;
+        const hasUUIDChanged = currentValue?.container.uuid !== previousValue?.container.uuid;
+
+        return hasIdentifierChanged || hasUUIDChanged;
+    }
+
```

Comment:
I don't think is responsibility of the contentlet to evaluate if something change here... I actually had the same problem and I solve it in the store: https://github.com/dotCMS/core/blob/49f0621cb4b492c5d10403a0e009aaa6eacb705c/core-web/libs/portlets/edit-ema/portlet/src/lib/dot-ema-shell/store/dot-ema.store.ts#L116-L131